src/features/import-maps.js

```diff
@@ -33,7 +33,22 @@ function processScripts () {
       script.sp = true;
       if (!script.src)
         return;
-      System.import(script.src.slice(0, 7) === 'import:' ? script.src.slice(7) : resolveUrl(script.src, baseUrl));
+      System.import(script.src.slice(0, 7) === 'import:' ? script.src.slice(7) : resolveUrl(script.src, baseUrl)).catch(function (e) {
+        // if there is a script load error, dispatch an "error" event
+        // on the script tag.
+        if (e.message.indexOf('https://git.io/JvFET#3') > -1) {
```

Comment:
Should we trigger `window.onerror` for the other cases similarly?

Reply:
According to my tests, when a browser encounters an error while handling a script tag it does the following:

- Error while fetching url behind `script.src`
-> Dispatch an `error` on the script tag
-> Logs the fetch error in the console

- Error while parsing code fetched from `script.src`
-> Dispatch an `error` on `window`
-> Logs syntax error in the console

- Error while executing code fetched from `script.src`
-> Dispatch an `error` on `window`
-> Logs the error in the console

As Systemjs uses a script tag, Browser is already dispatching syntax and runtime errors on `window`. We just need to handle the network error scenario to "redispatch" the `error` on the script tag.

Reply:
There is also MDN documentation that explain it better than me:

![image](https://user-images.githubusercontent.com/443639/98533644-873e4f80-2283-11eb-8796-39260f6c325a.png)

See https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror

Reply:
Of course it's already propogated yes, thanks for the clear explanation and investigation here - this sounds great.